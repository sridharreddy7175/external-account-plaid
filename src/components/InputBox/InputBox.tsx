import React from "react";
import { Box, Input } from "theme-ui";

export interface InputProps {
    name: string;
    type: string;
    value: string;
    onChange: any;
}

const InputBox = (props: InputProps) => {
    return (

        <Box pl={5} pr={5}>
            <Input
                name={props.name}
                type={props.type}
                mb={3}
                style={{ height: "45px" }}
                value={props.value}
                onChange={props.onChange}
            />

            {/* <p style={{ color: "red" }}>{emailError}</p> */}
        </Box>
    );
};

export default InputBox;
