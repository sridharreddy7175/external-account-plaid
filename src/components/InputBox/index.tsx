import InputBox from './InputBox';

export * from './InputBox';

export default InputBox;
