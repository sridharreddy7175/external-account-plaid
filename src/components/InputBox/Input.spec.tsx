import React from "react";
import InputBox from "./InputBox";
import renderer from "react-test-renderer";

describe("InputBox", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <InputBox type="box" name="email" value="email" onChange="OnChange" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export default null;
