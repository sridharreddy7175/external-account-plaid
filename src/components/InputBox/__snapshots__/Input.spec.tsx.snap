// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-i1z8pl{box-sizing:border-box;margin:0;min-width:0;padding-left:64px;padding-right:64px;}",
      }
    }
    data-emotion="css i1z8pl"
  />,
  <div
    className="css-i1z8pl"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-wcopiy{box-sizing:border-box;margin:0;min-width:0;display:block;width:100%;padding:8px;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;font-size:inherit;line-height:inherit;border:1px solid;border-radius:4px;color:inherit;background-color:transparent;margin-bottom:16px;}.css-wcopiy:autofill,.css-wcopiy:autofill:hover,.css-wcopiy:autofill:focus{box-shadow:inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);font-size:inherit;}.css-wcopiy:autofill:first-line,.css-wcopiy:autofill:hover:first-line,.css-wcopiy:autofill:focus:first-line{font-size:1rem;}.css-wcopiy:-webkit-autofill,.css-wcopiy:-webkit-autofill:hover,.css-wcopiy:-webkit-autofill:focus{box-shadow:inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);font-size:inherit;}.css-wcopiy:-webkit-autofill:first-line,.css-wcopiy:-webkit-autofill:hover:first-line,.css-wcopiy:-webkit-autofill:focus:first-line{font-size:1rem;}",
        }
      }
      data-emotion="css wcopiy"
    />
    <input
      className="css-wcopiy"
      name="email"
      onChange="OnChange"
      style={
        Object {
          "height": "45px",
        }
      }
      type="box"
      value="email"
    />
  </div>,
]
`;
